pygen/pygen_src/test/riscv_instr_base_test.py

```diff
@@ -12,22 +12,31 @@
 """
 
 import sys
-sys.path.append("../../")
-from pygen_src.riscv_instr_gen_config import cfg  # NOQA
+sys.path.append("pygen/")
+from pygen_src.riscv_instr_gen_config import cfg, args  # NOQA
 from pygen_src.isa.rv32i_instr import *  # NOQA
 from pygen_src.isa.riscv_instr import riscv_instr_ins  # NOQA
 from pygen_src.riscv_asm_program_gen import riscv_asm_program_gen  # NOQA
 
 
 class riscv_instr_base_test:
     def __init__(self):
-        pass
+        self.start_idx = args.start_idx
+        self.asm_file_name = args.asm_file_name
 
-    for _ in range(cfg.num_of_tests):
-        cfg.randomize()
-        asm = riscv_asm_program_gen()
-        riscv_instr_ins.create_instr_list(cfg)
-        test_name = "riscv_arithmetic_basic_test_{}.S".format(_)
-        asm.get_directed_instr_stream()
-        asm.gen_program()
-        asm.gen_test_file(test_name)
+    def run_phase(self):
+        for _ in range(cfg.num_of_tests):
+            cfg.randomize()
+            asm = riscv_asm_program_gen()
+            riscv_instr_ins.create_instr_list(cfg)
+            if cfg.asm_test_suffix != "":
+                self.asm_file_name = "{}.{}".format(self.asm_file_name,
+                                                    cfg.asm_test_suffix)
+            test_name = "{}_{}.S".format(self.asm_file_name,
+                                         _ + self.start_idx)
+            asm.get_directed_instr_stream()
+            asm.gen_program()
+            asm.gen_test_file(test_name)
+
+riscv_instr_base_test = riscv_instr_base_test()
+riscv_instr_base_test.run_phase()
```

Comment:
looks like you need to add newline at the end of file?

Reply:
I'll add the new line at the end of the file. However, it's not a big deal in python. Basically what it means is that the last byte in the file is not a newline, and it's for Diff purposes in git:  Diff has to output a newline anyway, or the result would be harder to read or process automatically.